Improvement/doc zenko 1206

wabernat opened the pull request:
This PR includes several improvements to the Installation Guide.

It fixes # ZENKO-1206

Most of the changes outside of the Installation Guide are to get the Ops Guide past make html.

Yes, I know it would be better if I used tox.

wabernat committed:
docs: fixes to the install guide

wabernat committed:
harmonized with ZENKO-1157 changes.

Added images for Orbit steps

Added admin.conf step for non-MK8s installers.

wabernat committed:
harmonized with ZENKO-1157 changes.

Added images for Orbit steps

Added admin.conf step for non-MK8s installers.

wabernat committed:
finished dogfooding. Final text changes (for now...)

incremental commit. html make won't build. rebasing.

incremental commit. (missed thi in the last one)

made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

Last loose ends before squashing commits.

incremental commit. html make won't build. rebasing.

incremental commit. (missed thi in the last one)

made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

turned off latex. Maybe it'll compile.

docs: adds installation step, clarifies, and fix typos

made changes named in ZENKO-1179

Also fixed note admonitions in a feeble attempt to avoid a merge conflict.

Updated the logo with new color scheme

ZENKO-1206 changes missed in last PR.

These were changes made/discovered in Flare source that had to be changed in RST source.

wabernat committed:
finished dogfooding. Final text changes (for now...)

wabernat committed:
finished dogfooding. Final text changes (for now...)

incremental commit. html make won't build. rebasing.

incremental commit. (missed thi in the last one)

made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

Last loose ends before squashing commits.

incremental commit. html make won't build. rebasing.

incremental commit. (missed thi in the last one)

made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

turned off latex. Maybe it'll compile.

docs: adds installation step, clarifies, and fix typos

made changes named in ZENKO-1179

Also fixed note admonitions in a feeble attempt to avoid a merge conflict.

Updated the logo with new color scheme

ZENKO-1206 changes missed in last PR.

These were changes made/discovered in Flare source that had to be changed in RST source.

wabernat committed:
finished dogfooding. Final text changes (for now...)

wabernat committed:
incremental commit. html make won't build. rebasing.

incremental commit. (missed thi in the last one)

made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

Last loose ends before squashing commits.

wabernat committed:
incremental commit. html make won't build. rebasing.

wabernat committed:
incremental commit. html make won't build. rebasing.

wabernat committed:
incremental commit. html make won't build. rebasing.

incremental commit. (missed thi in the last one)

made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

Last loose ends before squashing commits.

wabernat committed:
incremental commit. (missed thi in the last one)

wabernat committed:
made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

wabernat committed:
made some changes to the installation guide.

Had to make a lot of changes to the Operations Guide (beyond scope of ticket)
to make the project compile.

wabernat committed:
Merge branch 'improvement/doc-ZENKO-1206' of https://github.com/scality/Zenko into improvement/doc-ZENKO-1206

bert-e commented:
# Hello wabernatScality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/doc-ZENKO-1206` and the
destination branch `development/1.1`.

Please resolve the conflict on **the feature branch** (`improvement/doc-ZENKO-1206`).

```bash
 $ git fetch
 $ git checkout origin/improvement/doc-ZENKO-1206
 $ git merge origin/development/1.1
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/doc-ZENKO-1206
```

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/doc-ZENKO-1206` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`improvement/doc-ZENKO-1206`).

```bash
 $ git fetch
 $ git checkout origin/improvement/doc-ZENKO-1206
 $ git merge origin/development/1.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/doc-ZENKO-1206
```

wabernat committed:
Merge branch 'development/1.0' into improvement/doc-ZENKO-1206

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1206` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1206](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1206) will be merged into `development/1.1` (pull request #354)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @NicolasT

* @giacomoguiulfo

* @vrancurel

* @ssalaues

wabernat committed:
turned off latex. Maybe it'll compile.

wabernat committed:
turned off latex. Maybe it'll compile.

wabernat committed:
Merge branch 'improvement/doc-ZENKO-1206' of https://github.com/scality/Zenko into improvement/doc-ZENKO-1206

wabernat committed:
ZENKO-1206 changes missed in last PR.

These were changes made/discovered in Flare source that had to be changed in RST source.

wabernat committed:
ZENKO-1206 changes missed in last PR.

These were changes made/discovered in Flare source that had to be changed in RST source.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1206` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1206](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1206) will be merged into `development/1.1` (pull request #356)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @NicolasT

* @giacomoguiulfo

* @vrancurel

* @ssalaues

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1206` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1206](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1206) will be merged into `development/1.1` (pull request #357)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @NicolasT

* @giacomoguiulfo

* @vrancurel

* @ssalaues

anurag4DSB commented on docs/Operation-Architecture/Zenko_Architecture/Zenko_Service_Stack.rst:
```diff
@@ -1,97 +1,65 @@
+of Zenko’s structure. Several complexities are elided for clarity.
+
+Both transient source and NFS implementations are Transient source
+replication is optional and configurable. Transient source storage
```
Not sure if the first sentence is clear.

anurag4DSB replied:
Basically, I do not understand what we are trying to say in this line: "The Zenko instance depicted above presents an idealized representation of Zenko’s structure. Several complexities are elided for clarity.	of Zenko’s structure."

wabernat replied:
Looks like a couple of translation errors (Flare to RST) that got past me. I'll open a bug ticket for this.

anurag4DSB reviewed (dismissed):
Also please squash the commits :)

wabernat committed:
Merge branch 'improvement/doc-ZENKO-1206' of https://github.com/scality/Zenko into improvement/doc-ZENKO-1206

Trying to squash my commits only results in more commits...

ssalaues commented:
@wabernatScality it looks like there's some doc building errors

wabernat committed:
fixed index.rst for build.

wabernat committed:
fixed index.rst for build.

Commit:
Merge branch 'development/1.0' into improvement/doc-ZENKO-1206

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @vrancurel

* @NicolasT

* @giacomoguiulfo

* @ssalaues

ssalaues commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,75 +1,67 @@
-open-source Kubernetes cluster project, as described in "Setting Up a Metal K8s
-Cluster," below.
+Otherwise, you can set up a cluster quickly using
+`MetalK8s <https://github.com/scality/metal-k8s/>`__, Scality’s
```
While the old url will redirect for now it's best to switch all references of `metal-k8s` to `metalk8s` to prevent any issues in the future.

wabernat replied:
I agree. This may have been an older change coming in off a conflict resolution. As it stood last time I looked in on this, the unhyphenated URL was not resolving. Will address looking forward.

wabernat replied:
URL is fixed. Instances of the metalk8s venv still show the hyphen, so that remains in the documentation.

wabernat replied:
I don't know how to pull content review apart from "all these assets." I edited the docs while converting to RST. Sugar is cast upon the white sands. 

These changes have also been superseded, in part, by ZENKO-559. If you must be thorough, the best I can recommend is to limit scope to RST docs and leave the rest to tox. For this JIRA, the only significant changes will have been to docs/Installation/Installation.rst. The rest is nit-picking.

ssalaues reviewed (dismissed):
So far just a note about `metal-k8s` references. 

Also if possible, it would be easier to manage if we separated this into two PRs. One for the changes to the docs themselves and the other for all these assets being pulled in. Having all these assets in this PR makes it difficult to review.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/doc-ZENKO-1206` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`improvement/doc-ZENKO-1206`).

```bash
 $ git fetch
 $ git checkout origin/improvement/doc-ZENKO-1206
 $ git merge origin/development/1.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/doc-ZENKO-1206
```

wabernat committed:
Merge branch 'improvement/doc-ZENKO-1207' of https://github.com/scality/Zenko into improvement/doc-ZENKO-1206

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @vrancurel

* @NicolasT

* @ssalaues

wabernat committed:
ZENKO-1206. this builds. I have commits to squash.

ssalaues commented on eve/main.yml:
```diff
@@ -127,7 +127,7 @@ stages:
         name: 'Build doc'
         haltOnFailure: true
-        command: tox --workdir /tmp/tox -e docs -- html latexpdf
+        command: tox --workdir /tmp/tox -e docs -- html # latexpdf
```
Why are we disabling latexpdf in the CI? Is there some errors from the pdf generation?

Commit:
docs: import resources

bert-e commented:
# History mismatch



Merge commit #72a493e9d7c024ce7ae64f07b21479c40f5d5ac8 on the integration branch
`w/1.1/improvement/doc-ZENKO-1206` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1206` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1206` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ssalaues commented:
@bert-e force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1206` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1206](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1206) will be merged into `development/1.1` (pull request #384)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @vrancurel

* @NicolasT

* @ssalaues

ssalaues commented:
@anurag4DSB the commits have been squashed

wabernat committed:
Merge branch 'development/1.0' into improvement/doc-ZENKO-1206

wabernat committed:
Merge branch 'improvement/doc-ZENKO-1207' of https://github.com/scality/Zenko into improvement/doc-ZENKO-1206

wabernat committed:
Looked for conflicts. Found a bad heading.

fixed one remaining hyphenated "metal-k8s"

wabernat committed:
Looked for conflicts. Found a bad heading.

wabernat committed:
fixed one remaining hyphenated "metal-k8s"

wabernat committed:
Merge branch 'improvement/doc-ZENKO-1206' of https://github.com/scality/Zenko into HEAD

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/Zenko/commit/9192039970c2b610583bd67d1d149ab0c2af9f97) did not succeed in branch improvement/doc-ZENKO-1206.

Commit:
Merge branch 'development/1.0' into improvement/doc-ZENKO-1206

ssalaues commented:
What happened to the squashed commits? These were down to 2-3 commits

giacope reviewed (changes requested):
@wabernatScality We cannot let this in without squashing commits first. There are 35. Bert--e doesn't squash on merge (AFAIK), so the squashing has to be done manually. Please keep it less than 5 commits. Thanks

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @vrancurel

* @NicolasT

* @ssalaues

vrancurel commented:
Resolved by https://github.com/scality/Zenko/pull/412
